### PR TITLE
Terraform IaC for Azure PostgreSQL Single Server with Database and Private Endpoint

### PR DESCRIPTION
**Pull request by daniela-g-zheleva-db**
This PR is part of the epic https://github.com/finos/cloud-service-certification/issues/80. It consists of code to deploy PostgreSQL to Azure, in compliance with the security concerns described in the Service Accelerator Template already in the repo.

The tf code provisions a PostgreSQL Single Server with database and Private Endpoint. 

The tf is dependant on there being a virtual network already provisioned in the Azure environment which can be passed in as an input variable.

### CODE REVIEW

**Comment by mcleo-d** [comment]
Amazing pull request and contribution 🚀 

Thank you @peterrhysthomas, @daniela-g-zheleva-db, @fleadsom and Deutsche Bank ..! 

👏  💯 💯 💯

File: azure/postgresql/terraform/modules/postgres/variables.tf [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+  description = "Amount of storage for the database, in mb."
+  default = "5120"
+}
+#Number of days to retain backups
+variable "backup_retention_days" {
+  type = number
+  default = 7
+}
+#Enable geo redundency for Azure postgres
+#Enable geo redundancy for Azure postgres
```
Comment (eddie-knight) [inline azure/postgresql/terraform/modules/postgres/variables.tf:48]: nit: Duplicate line here

File: azure/postgresql/terraform/modules/postgres/variables.tf [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+variable "geo_redundent_enabled" {
+  type = bool
+  description = "Geo redundent postgres enablement, currently supports only: true"
```
Comment (eddie-knight) [inline azure/postgresql/terraform/modules/postgres/variables.tf:51]: I'm curious, why does this support only `true`? I noticed this below as well. Are these intended to be improved upon later- or why have a variable with only one supported value?

File: azure/postgresql/terraform/modules/postgres/variables.tf [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+}
+#Login for the database
+variable "database_login" {
+  type = string
+  description = "Login for server and database admin"
+}
+#Password for the database
+variable "database_password" {
+  type = string
+  description = "Password for the database in postgres"
```
Comment (eddie-knight) [inline azure/postgresql/terraform/modules/postgres/variables.tf:79]: Perhaps use the `sensitive` value for the credentials variables?
https://learn.hashicorp.com/tutorials/terraform/sensitive-variables#refactor-database-credentials

File: azure/postgresql/terraform/outputs.tf [hunk 4]
```diff
@@ -0,0 +1,16 @@
+output "server_name" {
+  value = "${var.postgres_name}"
```
Comment (eddie-knight) [inline azure/postgresql/terraform/outputs.tf:2]: nit: 
Perhaps this file doesn't need the interpolation syntax?
Possible improvement: `value = var.postgres_name`

**Review by eddie-knight — COMMENTED** [review]
This is great work. All of my comments are nits or minor suggestions.

Concerning the repo structure- perhaps it would be beneficial to segment the azure tf modules from their implementation logic? A common directory layout I've seen would look something like this:

```
azure/
  aks/
  postgresql/
  modules/
    aks/
    postgresql/
```

**Comment by daniela-g-zheleva-db** [comment]
Thanks @eddie-knight for the comments! I thought I had spotted all the 'porstgres' typos 😄  I will make these changes and push them into the PR as soon as possible.

**Comment by mcleo-d** [comment]
Thanks @eddie-knight and @daniela-g-zheleva-db for collaborating on this PR. It's great to see the Cloud Service Certification project coming together in such an affective way 👏  💯 💯 💯

cc @peterrhysthomas

**Comment by eddie-knight** [comment]
Per our conversation on the call today, I'm of the opinion that the best bet is to apply the suggested changes in a follow-up PR. FYI, whichever maintainer approves/merges this can use the Web UI to correct the `.gitignore` conflict.

**Comment by daniela-g-zheleva-db** [comment]
> Per our conversation on the call today, I'm of the opinion that the best bet is to apply the suggested changes in a follow-up PR. FYI, whichever maintainer approves/merges this can use the Web UI to correct the `.gitignore` conflict.

Thanks Eddy, @peter-thomas-db Could you please approve this PR and merge. Unfortunately I accidentally deleted the branch internally, and so cannot push changes to this PR. Once it is merged to main I will raise another PR to fix the spelling/spacing issues as pointed out by @eddie-knight

**Comment by mcleo-d** [comment]
👏 💯💯💯

**Pull request merged**